### PR TITLE
LibJS: More normative Temporal spec changes

### PR DESCRIPTION
**Pull request by linusg**
Once again not all of them affect us as a bunch of stuff isn't implemented in LibJS yet - turns out that's mostly stuff that still needed spec fixes. Lucky us :^)

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 1]
```diff
@@ -674,10 +674,20 @@ Optional<TemporalInstant> parse_temporal_instant_string(GlobalObject& global_obj
...
+    // 6. If timeZoneResult.[[Z]] is true, then
+    if (time_zone_result->z) {
+        // a. Set offsetString to "+00:00".
+        offset_string = "+00:00"sv;
+    }
+
+    // 7. Assert: offsetString is not undefined.
+    VERIFY(offset_string.has_value());
+
+    // TODO: This is supposed to use `offset_string`, see https://github.com/tc39/proposal-temporal/pull/1799
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:689]: I submitted a PR to fix this, if they merge it before CI turns green here I'll incorporate the change as well :)

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 2]
```diff
@@ -800,8 +810,8 @@ Optional<TemporalTimeZone> parse_temporal_time_zone_string(GlobalObject& global_
 
     // 4. If z is not undefined, then
     if (z_part.has_value()) {
-        // a. Return the Record { [[Z]]: "Z", [[OffsetString]]: "+00:00", [[Name]]: undefined }.
-        return TemporalTimeZone { .z = true, .offset = "+00:00", .name = {} };
+        // a. Return the Record { [[Z]]: true, [[OffsetString]]: undefined, [[Name]]: name }.
+        return TemporalTimeZone { .z = true, .offset = {}, .name = name_part.has_value() ? String { *name_part } : String {} };
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:814]: how about:
```c++
return TemporalTimeZone { .z = true, .offset = {}, .name = String { name_part.value_or({}) }};
```
I think it looks a bit nicer, but up to you

**Review by IdanHo — APPROVED** [review]
LGTM, just an optional stylistic suggestion:

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:814]: I don't think that's functionally equivalent, `Optional<T>::value_or(T&)` still returns a `T`, so in this case a null `StringView`, and I guess a null string - we want an empty optional...

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:814]: > we want an empty optional

oh no here we don't. good catch!

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:814]: > we want an empty optional

... and I messed that part up :^)

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp]: Changed `String {}` to `Optional<String> {}`. Still looks gross, but that's how `Optional<StringView>` -> `Optional<String>` conversion works, I suppose.

**Pull request merged**